File: src/main/java/com/clinica/atendimento/model/converters/DiaSemanaConverter.java
package com.clinica.atendimento.model.converters;

import com.clinica.atendimento.model.enums.DiaSemana;
import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

@Converter
public class DiaSemanaConverter implements AttributeConverter<DiaSemana, String> {

    @Override
    public String convertToDatabaseColumn(DiaSemana attribute) {
        if (attribute == null) {
            return null;
        }
        return attribute.getCodigo();
    }

    @Override
    public DiaSemana convertToEntityAttribute(String dbData) {
        if (dbData == null) {
            return null;
        }
        return DiaSemana.fromCodigo(dbData);
    }
}

File: src/main/kotlin/com/clinica/atendimento/model/converters/DiaSemanaConverter.kt
package com.clinica.atendimento.model.converters

import com.clinica.atendimento.model.enums.DiaSemana
import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class DiaSemanaConverter : AttributeConverter<DiaSemana, String> {

    override fun convertToDatabaseColumn(attribute: DiaSemana?): String? {
        return attribute?.codigo
    }

    override fun convertToEntityAttribute(dbData: String?): DiaSemana? {
        return if (dbData == null) null else DiaSemana.fromCodigo(dbData)
    }
}